src/typ.rs

```diff
@@ -777,6 +797,49 @@ impl<'a> Env<'a> {
         self.module_types.insert(name, info);
     }
 
+    pub fn get_module_real_type(&self, name: &str) -> Option<&Type> {
+        self.module_type_aliases
+            .get(name)
+            .map(|TypeAliasConstructor { ref typ, .. }| typ)
+    }
+
+    /// Map a type alias in the current scope.
+    ///
+    pub fn insert_type_alias(&mut self, name: String, info: TypeAliasConstructor) {
+        self.module_type_aliases.insert(name, info);
+    }
+
+    /// Lookup a type alias in the current scope.
+    ///
+    pub fn get_real_type(
```

Comment:
I find the term `real_type` a little unclear. Could we call this `resolve_type_alias`? Same for other places where `real_type` is used.

Reply:
maybe changing `get_real_type` to `resolve_type_alias` and `real_type` to `resolved_type` ?

Reply:
sounds good